tree/ntuple/inc/ROOT/RNTupleProcessorEntry.hxx

```diff
@@ -0,0 +1,266 @@
+   /// \param[in] valuePtr Pointer to the value for the field to add or update.
+   ///
+   /// \return `true` if the field was added successfully or if was already present and its value has been updated,
+   /// `false` otherwise.
+   ///
+   /// A field will only be updated if `valuePtr` is not a `nullptr`.
+   bool AddOrUpdateField(const std::string &fieldName, std::shared_ptr<void> valuePtr)
+   {
+      if (auto token = FindToken(fieldName)) {
+         if (valuePtr)
+            BindValue(*token, valuePtr);
+         return true;
+      }
+      ROOT::RFieldBase *field = fProtoModel->FindField(fieldName);
+      if (field) {
+         auto value = field->CreateValue();
+         if (valuePtr)
+            value.Bind(valuePtr);
+         fFieldName2Token[value.GetField().GetQualifiedFieldName()] = fValues.size();
+         fValues.emplace_back(std::move(value));
+         fFieldValidities.push_back(true);
+
+         return true;
+      }
+
+      return false;
+   }
+
+   /////////////////////////////////////////////////////////////////////////////
+   /// \brief Bind a new value pointer to a field in the entry.
+   ///
+   /// \param[in] fieldName The name of the field in the entry.
+   /// \param[in] objPtr Pointer to the value to bind to the field.
+   void BindValue(ROOT::RFieldToken token, std::shared_ptr<void> objPtr) { fValues[token.fIndex].Bind(objPtr); }
+
+   /////////////////////////////////////////////////////////////////////////////
+   /// \brief Bind a new value pointer to a field in the entry.
+   ///
+   /// \param[in] fieldName The name of the field in the entry.
+   /// \param[in] objPtr Pointer to the value to bind to the field.
+   void BindValue(std::string_view fieldName, std::shared_ptr<void> objPtr)
+   {
+      auto token = FindToken(fieldName);
+      if (!token)
+         throw RException(R__FAIL("invalid field name: " + std::string(fieldName)));
+      BindValue(*token, objPtr);
+   }
+
+   /////////////////////////////////////////////////////////////////////////////
+   /// \brief Read the field values for the entry corresponding to the provided index.
+   ///
+   /// \param[in] index The entry number to read.
+   void Read(ROOT::NTupleSize_t index)
+   {
+      for (unsigned i = 0; i < fValues.size(); i++) {
+         if (fFieldValidities[i])
+            fValues[i].Read(index);
+      }
+   }
+
+   /////////////////////////////////////////////////////////////////////////////
+   /// \brief Get a pointer to the value for the field represented by the provided token.
+   ///
+   /// \tparam T The type of the pointer
+   ///
+   /// \param[in] token The token representing the field in the entry.
+   template <typename T>
+   std::shared_ptr<T> GetPtr(ROOT::RFieldToken token) const
+   {
+      assert(token.fSchemaId == fProtoModel->fSchemaId && token.fIndex < fValues.size());
+
+      if (fFieldValidities[token.fIndex])
+         return std::static_pointer_cast<T>(fValues[token.fIndex].GetPtr<void>());
```

Comment:
Isn't this potentially dangerous without a type check? Should it do
```suggestion
         return fValues[token.fIndex].GetPtr<T>();
```
(knowing that right now, `RValue::GetPtr` doesn't do a type check either - we should fix https://github.com/root-project/root/issues/18316...)